### DISCUSSION TITLE
How to get the channel name in slash command

### DISCUSSION THREAD

**Discussion by nico-qwer** [Rapptz/discord.py#9167] [Q&A] [answered]
Because I am changing my bot to slash commands, I changed the context in normal text commands for `discord.Interaction`.  
However, one of my commands used `ctx.channel`, which I can't find in `discord.Interaction`.  
I tried using `discord.Interaction.channel`, but i got this error:  
`TypeError: 'TextChannel' object is not callable`  

How can I achieve this?

**Comment by benz206** [comment, marked as answer]
The object `discord.Interaction` does have an attribute called channel, you just can't call it directly without instantiating an object...

You can view the documentation [here](https://discordpy.readthedocs.io/en/stable/interactions/api.html?highlight=discord%20interaction#discord.Interaction.channel).

![image](https://user-images.githubusercontent.com/57199957/211440515-e931509e-baf3-4625-8006-04bc25e431c7.png)
It should also be noted that this could be `None`.

**Reply by nico-qwer** [reply]
So how would I get the name of the channel the message was sent in?
```
@bot.tree.command()
async def command_name(inter: discord.Interaction):
    channel_name = ?????????
```

**Reply by AbstractUmbra** [reply]
`channel_name = inter.channel.name`.
